systemtest/src/test/java/io/strimzi/systemtest/operators/LeaderElectionIsolatedST.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * The whole procedure of deploying CO with Leader Election enabled and many more is described in
+ *
+ * <a href="https://strimzi.io/docs/operators/in-development/configuring.html#assembly-using-multiple-cluster-operator-replicas-str">the documentation</a>
+ */
+
+@Tag(REGRESSION)
+@IsolatedSuite
+public class LeaderElectionIsolatedST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(LeaderElectionIsolatedST.class);
+
+    /*
+     These env variables are already inside packaging/../060-Deployment-strimzi-cluster-operator.yaml, but to be sure that
+     we are setting everything correctly and the Leader Election is really enabled, we are adding it to additional envs
+     */
+    final EnvVar leaderEnabledEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_ENABLED")
+        .withValue("true")
+        .build();
+
+    final EnvVar leaseNameEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAME")
+        .withValue(Constants.STRIMZI_DEPLOYMENT_NAME)
+        .build();
+
+    final EnvVar leaseNamespaceEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.namespace")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final EnvVar leaderIdentityEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_IDENTITY")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.name")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final List<EnvVar> envList = Arrays.asList(leaderEnabledEnv, leaseNameEnv, leaseNamespaceEnv, leaderIdentityEnv);
+
+    @IsolatedTest
+    void testLeaderElection(ExtensionContext extensionContext) {
+        TestStorage testStorage = new TestStorage(extensionContext);
+
+        clusterOperator = clusterOperator.defaultInstallation()
+            .withExtensionContext(extensionContext)
+            .withExtraEnvVars(envList)
+            .withReplicas(2)
+            .createInstallation()
+            .runInstallation();
+
+        Lease oldLease = kubeClient().getClient().leases().inNamespace(testStorage.getNamespaceName()).withName(Constants.STRIMZI_DEPLOYMENT_NAME).get();
+        String oldLeaderPodName = oldLease.getSpec().getHolderIdentity();
+
+        LOGGER.info("Changing image of the leader pod: {} to not available image - to cause CrashLoopBackOff and change of leader to second pod (failover)", oldLeaderPodName);
+
+        kubeClient().editPod(testStorage.getNamespaceName(), oldLeaderPodName).edit(pod -> new PodBuilder(pod)
+            .editOrNewSpec()
+                .editContainer(0)
+                    .withImage("wrong-image/name:latest")
+                .endContainer()
+            .endSpec()
+            .build()
+        );
```

Comment:
So, you have 2 replicas. And when you change the image, you expect that one will be deleted and the new pod will not start because of faulty image. But how do you know that it will be the leader who is deleted and fails to start and not the follower?

Reply:
I have the `oldLeaderPodName` variable, which holds the name of the pod which was leader before. So I'm editing pod which _is_ leader actually. That way we'll know that it will move to the second pod.

Reply:
Ahh, ok. So ... you edit the Pod directly under the Deployments hands? I wonder how stable that is, but I guess as long as it works 🤷‍♂️

Reply:
I let it be like 10 minutes and it was in the crash, I thought K8s will roll it somehow. We'll see how it will work on Azure